### PR TITLE
No need to install SDL into the containers

### PR DESCRIPTION
**Pull request by incandescant**
Since Krogoth (2.1), or perhaps earlier, sdl-native is built by the
build system rather than relying on the host to have SDL headers
installed. Therefore save some time and space in the containers by
removing SDL from the list of installed dependencies.

Signed-off-by: Joshua Lock <joshua.g.lock@intel.com>

### CODE REVIEW

**Comment by rewitt1** [comment]
Joshua,

I tested this locally and got the same failure as on travis. build_container.sh also runs the tests, so if you wanted to try it locally you can run.

```
REPO=deleteme DISTRO_TO_BUILD=ubuntu-14.04 ./build_container.sh
```

However, even if it worked, don't we still support 2.0, 2.1, 2.2? And it is a good question whether we actively want to prevent the containers from working on older releases.

I always wanted to aggressively prune the images, but it would be nice to do it without removing things older releases explicitly require.

**Comment by incandescant** [comment]
The core project team are only supporting 2.1 & 2.2. Older releases are now supported by the community.
jethro/2.0 and newer branches support building sdl-native (we backported the patches) and don't default to adding sdl-native to ASSUME_PROVIDED. 

I'd argue the overhead of sdl (including an X stack) aren't worth it to support unmaintained branches that were originally release almost 2 years ago.

If the CROPS team agree I can work on resolving the failure.

**Comment by incandescant** [comment]
This series has been updated to:

1. remove SDL from the container images
2. add the recently released opensuse 42.3
3. remove all of the EOL'd distro versions

Please consider if for merge.

**Comment by rewitt1** [comment]
Re: Removing the EOL'd distros.

Although not required until the removal of the images from dockerhub, we also need a corresponding change to crops/poky.

****RFC**** @moto-timo @bavery22 
Tim and I were just talking about this a couple of days ago. The question I have is, if we no longer build them, should we still leave the last successfully generated images on dockerhub?

Because in some cases the images may no longer be able to be generated. For instance, this happened with opensuse 13.2, due to the removal updates repo.

I think I lean toward not even preserving the old images on dockerhub, since we're not really an archival service, and instead if a user needs that, it is up to them to preserve whatever images they might need.

However, I wanted to let other people chime is as well.
****RFC****

File: dockerfiles/opensuse/opensuse-42.3/opensuse-42.3-builder/README [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The Dockerfile for this directory is autogenerated based on dockerfiles/templates/Dockerfile.builder
```
Comment (rewitt1) [inline dockerfiles/opensuse/opensuse-42.3/opensuse-42.3-builder/README]: The builder directories were completely removed with a previous commit. So there isn't a need to add this anymore.

Comment (incandescant) [inline dockerfiles/opensuse/opensuse-42.3/opensuse-42.3-builder/README]: I'd missed that, fix pushed.

**Comment by bavery22** [comment]
I heartily endorse the idea of retiring/removing the non buildable versions esp since those won't get updates...

**Pull request merged**